### DISCUSSION TITLE
Decoding Dependent picklists "validfor" values?

### DISCUSSION THREAD

**Discussion by mbarzeev** [jsforce/jsforce#1492] [Q&A]
Hi,
We're getting Picklist values from the field's describe, but as you know some Picklists are dependent on others to filter their options ("controllerName").
The filter is done using a base64 encoded bitmap each Picklist value has ("validFor").
I'm trying to figure out if there is an easy way to decode this bitmap with your library. I couldn't find a way to do so :(
Are there any best-practices you would recommend on doing that?

Thanks

**Comment by Hellwest** [comment]
I struggle with this, too...

**Comment by paustint** [comment]
You could possibly look into the UI API, which might provide better responses for this (but might still have other complications): https://developer.salesforce.com/docs/atlas.en-us.uiapi.meta/uiapi/ui_api_features_records_dependent_picklist.htm

Since there are no built ins for calling the UI api, you would need to use the method mentioned here: https://github.com/jsforce/jsforce/issues/934#issuecomment-546714813
